R/rowClust.R

```diff
@@ -0,0 +1,41 @@
+# REQUIRED PACKAGES: "fastcluster", "stats", "coop"
+rowClust <- function(tse,
+                     abund_values = "counts",
+                     metric = "spearman", # c("pearson", "kendall", "spearman" || 
+                                          #    "cosine" ||
+                                          #    "euclidean", "maximum", "manhattan", "canberra", "binary", "minkowski")
+                     method = "ward.D2", # c("single", "complete", "average", "mcquitty", "ward.D", "ward.D2", "centroid","median")
+                     name = "clust_id",
+                     k = NULL, # number of clusters
+                     h = NULL, # height of cut-off
+                     ... # arguments passed to stats::dist (mainly for the p arg of "minkowski")
+                     ) {
+  # Getting Assay data
+  if(abund_values %in% assayNames(tse))
+    x <- assay(tse, abund_values)
+  else
+    stop(paste0("Assay ",abund_values," not Found!"))
+  
+  # Calculating distance
+  if (metric %in% c("pearson", "kendall", "spearman"))
+    x <- as.dist(cor(t(x), method = metric))
+  else if (metric=="cosine")
+    x <- as.dist(1 - coop::cosine(t(x)))
+  else if (metric %in% c("euclidean", "maximum", "manhattan", "canberra", "binary", "minkowski"))
```

Comment:
In general, in `mia` we use helper functions so that tests and checking bugs are easier. 
You can create helper functions for these options and also use `match.arg()` and  `switch()` for example used here https://github.com/microbiome/mia/blob/master/R/summaries.R